### PR TITLE
sops: various improvements and tests

### PR DESCRIPTION
**Pull request by hiddeco**
This PR revises the SOPS keysource implementations, removes dead code, and adds tests.

### CODE REVIEW

File: internal/sops/keyservice/server.go [hunk 1]
```diff
@@ -47,20 +41,101 @@ type Server struct {
...
 		},
 	}
 	return server
 }
 
+// Encrypt takes an encrypt request and encrypts the provided plaintext with
+// the provided key, returning the encrypted result.
+func (ks Server) Encrypt(ctx context.Context, req *keyservice.EncryptRequest) (*keyservice.EncryptResponse, error) {
+	key := req.Key
+	switch k := key.KeyType.(type) {
```
Comment (pjbgf) [inline internal/sops/keyservice/server.go:61]: May want to add a `nil` check for `req` and a default case for the `switch`.

File: internal/sops/keyservice/server.go [hunk 2]
```diff
@@ -47,20 +41,101 @@ type Server struct {
...
+	}
+	// Fallback to default server for any other request.
+	return ks.DefaultServer.Encrypt(ctx, req)
+}
+
+// Decrypt takes a decrypt request and decrypts the provided ciphertext with
+// the provided key, returning the decrypted result.
+func (ks Server) Decrypt(ctx context.Context, req *keyservice.DecryptRequest) (*keyservice.DecryptResponse, error) {
+	key := req.Key
+	switch k := key.KeyType.(type) {
```
Comment (pjbgf) [inline internal/sops/keyservice/server.go:97]: Same here

File: internal/sops/age/keysource.go [hunk 3]
```diff
@@ -14,102 +14,159 @@ import (
...
+	// EncryptedKey contains the SOPS data key encrypted with age.
+	EncryptedKey string
+
+	// parsedIdentities contains a slice of parsed age identities.
+	// It is used to lazy-load the Identities once.
+	// It can also be injected by a (local) keyservice.KeyServiceServer using
+	// ParsedIdentities.ApplyToMasterKey().
+	parsedIdentities []age.Identity
+	// parsedRecipient contains a parsed age public key.
+	// It is used to lazy-load the Recipient once.
```
Comment (pjbgf) [inline internal/sops/age/keysource.go:41]:
```suggestion
	// It is used to lazy-load the Recipient at-most once.
```

File: internal/sops/age/keysource.go [hunk 4]
```diff
@@ -14,102 +14,159 @@ import (
...
+	// In addition to using this field, ParsedIdentities.ApplyToMasterKey() can
+	// be used to lazy-load parse and lazy-load identities.
+	Identities []string
+	// Recipient contains the Bench32-encoded age public key used to Encrypt.
+	Recipient string
+	// EncryptedKey contains the SOPS data key encrypted with age.
+	EncryptedKey string
+
+	// parsedIdentities contains a slice of parsed age identities.
+	// It is used to lazy-load the Identities once.
```
Comment (pjbgf) [inline internal/sops/age/keysource.go:36]:
```suggestion
	// It is used to lazy-load the Identities at-most once.
```

**Review by pjbgf — APPROVED** [review]
@hiddeco :top::top: stuff!

LGTM

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @hiddeco 🥇

**Pull request merged**